code-studio/src/js/components/progress/course_progress.jsx

```diff
@@ -2,15 +2,25 @@
 
 var STAGE_TYPE = require('./types').STAGE_TYPE;
 var CourseProgressRow = require('./course_progress_row');
+var StageDetails = require('./stage_details');
 
 /**
  * Stage progress component used in level header and course overview.
  */
 var CourseProgress = React.createClass({
   propTypes: {
+    display: React.PropTypes.oneOf(['dots', 'list']).isRequired,
     stages: React.PropTypes.arrayOf(STAGE_TYPE)
   },
 
+  getRow: function (stage) {
+    if (this.props.display === 'dots') {
+      return <CourseProgressRow stage={stage} key={stage.name}/>;
+    } else {
+      return <StageDetails  stage={stage} key={stage.name}/>;
```

Comment:
My personal preference is no space. I also don't see a lot of harm in allowing both.